test/rx_ext/repeater_test.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+
+import 'package:flutter_reactive_ble/src/rx_ext/repeater.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+void main() {
+  group("Repeater", () {
+    StreamController<int> underlyingStreamController;
+    _RepeaterHandlerMock<int> handler;
+    Repeater<int> sut;
+
+    setUp(() {
+      underlyingStreamController = StreamController(sync: true);
+      handler = _RepeaterHandlerMock();
+
+      when(handler.onListenEmitFrom())
+          .thenAnswer((_) => underlyingStreamController.stream);
+    });
+
+    tearDown(() {
+      underlyingStreamController.close();
+    });
+
+    group('Repeater internals', () {
+      setUp(() {
+        sut = Repeater(
+          onListenEmitFrom: handler.onListenEmitFrom,
+          onCancel: handler.onCancel,
+          isSync: true,
+        );
+      });
```

Comment:
_Usually_ (not saying "always"), it's possible to initialize the SUT _once_ in the top-level group. _Usually_ it takes just a bit more mocking (not sure it's the case here).

Reply:
I use different constructors for the sut (regular, fromStream and fromBroadcast).  I declared the sut in the respective group so test is more readable now

Reply:
That may be considered a wrong abstraction: it saves you some typing, but causes questions, thus not being sound. Yes, I mean being honest and declaring the SUT in multiple groups would be more straightforward and unquestionable. No, I don't insist too much on changing this.